analytics-integrations/taplytics/src/main/java/com/segment/analytics/internal/integrations/TaplyticsIntegration.java

```diff
@@ -0,0 +1,127 @@
+    static final String TAPLYTICS_KEY = "Taplytics";
+
+    static final Set<String> knownTraitNames = new HashSet<String>() {{
+        add("name");
+        add("email");
+        add("gender");
+        add("firstName");
+        add("lastName");
+        add("age");
+    }};
+
+    @Override
+    public void initialize(Analytics analytics, ValueMap settings) throws IllegalStateException {
+        String apiKey = settings.getString("appkey");
+        settings.remove("appkey");
+        HashMap options = new HashMap();
+
+        for (Map.Entry<String, String> entry : settings.toStringMap().entrySet()) {
+            if (entry.getValue().toLowerCase().equals("true") || entry.getValue().toLowerCase().equals("false")) {
+                options.put(entry.getKey(), Boolean.valueOf(entry.getValue()));
+            } else {
+                options.put(entry.getKey(), entry.getValue());
+            }
+        }
+        Taplytics.startTaplytics(analytics.getApplication(), apiKey, options);
+    }
+
+    @Override
+    public String key() {
+        return TAPLYTICS_KEY;
+    }
+
+    @Override
+    public void identify(IdentifyPayload identify) {
+        super.identify(identify);
+        JSONObject userAttributes = new JSONObject();
+        JSONObject customData = new JSONObject();
+        JSONObject traits = identify.traits().toJsonObject();
+        for (Iterator<String> jsonIterator = traits.keys(); jsonIterator.hasNext(); ) {
+            String key = jsonIterator.next();
+            try {
+                Object value = traits.get(key);
+
+                if (knownTraitNames.contains(key) && !Utils.isNullOrEmpty(value.toString())) {
+                    userAttributes.put(key, value);
+                } else if (key.equals("userId") && !Utils.isNullOrEmpty(value.toString())) {
+                    userAttributes.put("user_id", value);
+                } else {
+                    customData.put(key, value);
+                }
+            } catch (JSONException e) {
+                //Ignore this value if it is problematic.
+            }
+        }
+
+        try {
+            if (customData.length() > 0) {
+                userAttributes.put("customData", customData);
+            }
+        } catch (JSONException e)
+
+        {
+            //We will just leave out the custom data if its problematic.
+        }
+
+        Taplytics.setUserAttributes(userAttributes);
+    }
+
+    @Override
+    public void track(TrackPayload track) {
+        super.track(track);
+        JSONObject metaData = track.properties().toJsonObject();
+        Taplytics.logEvent(track.event(), null, metaData);
+    }
+
+
+    @Override
+    public void group(GroupPayload group) {
+        super.group(group);
+
+        JSONObject userAttributes = new JSONObject();
+        JSONObject customData = group.traits().toJsonObject();
+
+        try {
+            userAttributes.put("customData", customData);
```

Comment:
Is there a reason to put this under `customData` instead of something like group?